bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/utils/Utils.java

```diff
@@ -156,4 +165,131 @@ public static ZonedDateTime getZdtFromUTC(Instant utc) {
     public static Instant now() {
         return Instant.now(clock);
     }
+
+    /**
+     * Check if an item has historic data in the persistence service
+     *
+     * @param item the item name to check
+     * @param service the persistence service to query
+     * @return true if there is historic data for the item, false otherwise
+     */
+    public static boolean checkPersistence(String item, QueryablePersistenceService service) {
+        return getEnergyTillNow(item, service).isPresent();
+    }
+
+    /**
+     * Get the energy produced by the calculation item since the beginning of the current day
+     *
+     * @param calculationItem the name of the power / energy item
+     * @param service the persistence service to query
+     * @return the total energy produced in kWh, empty if the item unit is not power or energy
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
Pleasew remove these kind of suppresions

Reply:
removed

Reply:
Removed all SuppressWarnings